docs/metrics/geometry-matcher.md

```diff
@@ -0,0 +1,275 @@
+---
+search:
+  exclude: true
+---
+
+# Geometry Matcher
+
+Geometry matcher finds the best possible match, given the sets of the ground truth and prediction polygons for each
+image. The geometry matcher utility is a building block for any object detection metrics, such as precision, recall,
+and average precision. In this guide, we will focus on 2D object detection workflow, specifically 2D bounding box
+matching.
+
+The criteria for a valid match is as follows:
+
+1. The prediction label matches the ground truth label.
+2. The prediction polygon has more than or equal to a certain overlap (i.e. IoU threshold) with the ground truth polygon.
+
+Here is the general matching logic:
+
+1. Loop through all images in your dataset;
+2. Loop through all labels;
+3. Get predictions and ground truths with the evaluating label;
+4. Sort predictions by descending confidence score;
+5. Check against all ground truths and find a ground truth that results in maximum IoU;
+6. Check for the following criteria for a valid match:
+	1. This ground truth is not matched yet AND
+	2. The IoU is greater than or equal to the IoU threshold;
+7. Repeat 5-6 on the next prediction;
```

Comment:
Do you think this provides value, @y27choi? Would a reader of these docs care?

Reply:
For me personally, it helps me understand the logic a bit better with the pseudocode.  It was easier to highlight the difference when comparing matching algorithm variations.  I learned this matching algorithm so I can implement it so my view on this could be biased.

I take that you don't find it useful?  Is it more noise to you?  Will it make it better if I hide in in a collapsible?

Reply:
Put in a collapsible block.